hail/python/test/hail/test_ir.py

```diff
@@ -357,3 +359,79 @@ def test_value_same_after_parsing(self):
                     None))
             new_globals = hl.eval(hl.Table(map_globals_ir).index_globals())
             self.assertEqual(new_globals, hl.Struct(foo=v))
+
+
+class CSETests(unittest.TestCase):
+    def test_cse(self):
+        x = ir.I32(5)
+        x = ir.ApplyBinaryPrimOp('+', x, x)
+        expected = (
+            '(Let __cse_1 (I32 5)'
+            ' (ApplyBinaryPrimOp `+`'
+                ' (Ref __cse_1)'
+                ' (Ref __cse_1)))')
+        self.assertEqual(expected, CSERenderer()(x))
```

Comment:
I have a slight preference for pytest assertions: `assert expected == CSERenderer()(x)`

They're visually cleaner and I think they generate slightly better errors

Reply:
Ah, I didn't realize there were different assertions!